lib/msf/base/simple/exploit.rb

```diff
@@ -186,7 +186,7 @@ def self.check_simple(mod, opts, job_listener: Msf::Simple::NoopJobListener.inst
 
     unless mod.has_check?
       # Bail out early if the module doesn't have check
-      raise Msf::ValidationError, Msf::Exploit::CheckCode::Unsupported.message
+      raise ::NoMethodError.new(Msf::Exploit::CheckCode::Unsupported.message, 'check')
```

Comment:
This error made more sense to me 🤔

Reply:
I don't think anyone else in the codebase has used `Msf::ValidationError` before, at least from a quick code search that appears to be the case

https://github.com/search?l=Ruby&q=org%3Arapid7+ValidationError&type=Code

Reply:
Yeah, that's better.

Reply:
(Since this isn't an `ArgumentError` like `Msf::OptionValidateError` and more a `NameError`.)